dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -1,27 +1,101 @@
-// Copyright 2023 ChainSafe Systems (ON)
+// Copyright 2024 ChainSafe Systems (ON)
 // SPDX-License-Identifier: LGPL-3.0-only
 
-package parachain
+package candidatevalidation
 
 import (
 	"bytes"
+	"context"
 	"errors"
 	"fmt"
+	"sync"
 
 	parachainruntime "github.com/ChainSafe/gossamer/dot/parachain/runtime"
 	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/internal/log"
 	"github.com/ChainSafe/gossamer/lib/common"
-
 	"github.com/ChainSafe/gossamer/pkg/scale"
 )
 
+var logger = log.NewFromGlobal(log.AddContext("pkg", "parachain-candidate-validation"))
+
 var (
 	ErrValidationCodeMismatch   = errors.New("validation code hash does not match")
 	ErrValidationInputOverLimit = errors.New("validation input is over the limit")
 )
 
+type CandidateValidation struct {
+	wg       sync.WaitGroup
+	stopChan chan struct{}
+
+	SubsystemToOverseer chan<- any
+	OverseerToSubsystem <-chan any
+}
+
+func NewCandidateValidation(overseerChan chan<- any) *CandidateValidation {
+	candidateValidation := CandidateValidation{
+		SubsystemToOverseer: overseerChan,
+	}
+
+	return &candidateValidation
+}
+
+func (cv *CandidateValidation) Run(ctx context.Context, OverseerToSubSystem chan any, SubSystemToOverseer chan any) {
+	cv.wg.Add(1)
+	go cv.processMessages(&cv.wg)
+}
+
+func (*CandidateValidation) Name() parachaintypes.SubSystemName {
+	return parachaintypes.CandidateValidation
+}
+
+func (*CandidateValidation) ProcessActiveLeavesUpdateSignal(signal parachaintypes.ActiveLeavesUpdateSignal) error {
```

Comment:
As we are not using this function argument. Let's remove it.
```suggestion
func (*CandidateValidation) ProcessActiveLeavesUpdateSignal(parachaintypes.ActiveLeavesUpdateSignal) error {
```

Reply:
Removed.